aspnetcore/fundamentals/middleware/request-decompression.md

```diff
@@ -0,0 +1,83 @@
+---
+title: Request Decompression in ASP.NET Core
+author: david-acker
+description: Learn how to use the request decompression middleware in ASP.NET Core
+monikerRange: '>= aspnetcore-7.0'
+ms.author: riande
+ms.date: 8/17/2022
+uid: fundamentals/middleware/request-decompression
+---
+# Request decompression in ASP.NET Core
+
+By [David Acker](https://github.com/david-acker)
+
+Request decompression middleware:
+
+* Enables API endpoints to accept requests with compressed content.
+* Uses the [`Content-Encoding`](https://developer.mozilla.org/docs/Web/HTTP/Headers/Content-Encoding) HTTP header to automatically identify and decompress requests which contain compressed content.
+* Eliminates the need to write code to handle compressed requests.
+
+When the `Content-Encoding` header value on a request matches one of the available decompression providers, the middleware:
+
+* Uses the matching provider to wrap the <xref:Microsoft.AspNetCore.Http.HttpRequest.Body?displayProperty=nameWithType> in an appropriate decompression stream.
+* Removes the `Content-Encoding` header, indicating that the request body is no longer compressed.
+
+Requests that don't include a `Content-Encoding` header are ignored by the request decompression middleware.
+
+Decompression:
+
+* Occurs when the body of the request is being read. That is, decompression occurs at the endpoint on model binding. The request body is not decompressed eagerly.
+* When attempting to read the decompressed request body, if the compressed data is invalid for the specified `Content-Encoding`, an exception is thrown.
```

Comment:
What kind of exception? Since this is Stream, only a few types of exceptions are expected, like IOException.

Reply:
These are the exceptions thrown by the underlying decompression stream implementations used by the default providers:

**Brotli:**
System.InvalidOperationException: Decoder ran into invalid data.

**Deflate:**
System.IO.InvalidDataException: The archive entry was compressed using an unsupported compression method.

**GZip:**
System.IO.InvalidDataException: The archive entry was compressed using an unsupported compression method.